File: max_profit.c
// max_profit.c
#include <stdio.h>
#include <limits.h>

int maxProfit(int prices[], int size) {
    if (size < 2) return 0;

    int p = 0, mini = prices[0];
    for (int i = 1; i < size; i++) {
        if (prices[i] == INT_MIN) continue;  // Ignore invalid prices
        if (prices[i] < mini) mini = prices[i];
        if (prices[i] - mini > p) p = prices[i] - mini;
    }
    return p;
}

File: max_profit.java
public class max_profit {
    public int maxProfit(int[] prices) {
        if (prices.length < 2) return 0; // Edge case: If only one price, no transaction possible

        int p = 0, mini = prices[0];
        for (int i = 1; i < prices.length; i++) {
            mini = Math.min(mini, prices[i]);
            p = Math.max(p, prices[i] - mini);
        }
        return p;
    }
}
